python/setup.py

```diff
@@ -227,6 +228,126 @@ def initialize_options(self):
         '_hdfsio',
         'gandiva']
 
+    def _run_cmake_pyarrow_cpp(self):
+        # check if build_type is correctly passed / set
+        if self.build_type.lower() not in ('release', 'debug'):
```

Comment:
For the record, can we open a JIRA to also support `RelWithDebInfo`?

Reply:
Done: https://issues.apache.org/jira/browse/ARROW-17488